modules/check_in/app/controllers/check_in/v2/demographics_controller.rb

```diff
@@ -7,11 +7,26 @@ class DemographicsController < CheckIn::ApplicationController
       after_action :after_logger, only: %i[update], if: :additional_logging?
 
       def update
-        head :not_implemented
+        check_in_session = CheckIn::V2::Session.build(data: { uuid: params[:id] }, jwt: session[:jwt])
+
+        unless check_in_session.authorized?
+          render json: check_in_session.unauthorized_message, status: :unauthorized and return
+        end
+
+        if permitted_params[:demographic_confirmations]
+          resp = ::V2::Chip::Service.build(check_in: check_in_session,
+                                           params: permitted_params[:demographic_confirmations])
+                                    .confirm_demographics
+          render json: resp
+        else
+          head :not_implemented
```

Comment:
We should not be returning 501 based on the request parameters validation. 501 basically means that the HTTP method is not implemented currently, and in this case that is not correct. Better response would be a 400 Bad Request if request parameters fail validation.

Reply:
Addressed in [5446e](https://github.com/department-of-veterans-affairs/vets-api/pull/9115/commits/5446eecf8bcf5b1c6b6aed246880bd83ea6ada56)

Reply:
As discussed offline, moved the parameter check to service layer in [commit](https://github.com/department-of-veterans-affairs/vets-api/pull/9115/commits/cad945c87e1a5a996cff3871f21f5788ac58cabf)